packages/osd-stylelint-plugin-stylelint/src/utils/get_rules_from_config.ts

```diff
@@ -18,7 +18,7 @@ export const getRulesFromConfig = (configPath: string) => {
   return JSON.parse(readFileSync(filePath, 'utf-8'));
 };
 
-export const getSelectorRule = (rules: any, rule: any) => {
+export const getSelectorRule = (rules: Record<string, { approved?: string[] }>, rule: any) => {
```

Comment:
nit: might be a little bit over-engineering but we to define a `type` for the two rules config types we have and import and export them within the package?

for example we have a rules config that is files based so in a `index.d.ts` somewhere in the stylelint plugin package:

```typescript
export type FileBasedConfig = Record<string, { approved?: string[] }>;
export type ValueBasedConfig = Record<
  string,
  Record<string, Array<{ approved?: string; rejected?: string[] }>>
>;
```

Reply:
Yep, adding a type for `ValueBasedConfig` also exposed a few cases when things could have been null but weren't tested against